cli/common/utils.go

```diff
@@ -143,3 +145,91 @@ func ParseYmlFile(path string) (map[string]interface{}, error) {
 
 	return res, nil
 }
+
+func readFromPos(f *os.File, pos int64, buf *[]byte) error {
+	if _, err := f.Seek(pos, io.SeekStart); err != nil {
+		return fmt.Errorf("Failed to seek: %s", err)
+	}
+
+	if _, err := f.Read(*buf); err != nil {
+		return fmt.Errorf("Failed to read: %s", err)
+	}
+
+	return nil
+}
+
+// GetLastNLinesBegin return the position of last n lines begin
+func GetLastNLinesBegin(filepath string, n int) (int64, error) {
+	if n == 0 {
+		return 0, nil
+	}
+
+	f, err := os.Open(filepath)
+	if err != nil {
+		return 0, fmt.Errorf("Failed to open log file: %s", err)
+	}
+	defer f.Close()
+
+	var fileSize int64
+	if fileInfo, err := os.Stat(filepath); err != nil {
+		return 0, fmt.Errorf("Failed to get fileinfo: %s", err)
+	} else {
+		fileSize = fileInfo.Size()
+	}
+
+	if fileSize == 0 {
+		return 0, nil
+	}
+
+	buf := make([]byte, bufSize)
+
+	var filePos int64 = fileSize - bufSize
+	var lastNewLinePos int64 = 0
+	var newLinesN int = 0
+
+	// check last symbol of the last line
+
+	if err := readFromPos(f, fileSize-1, &buf); err != nil {
+		return 0, fmt.Errorf("%s", err)
+	}
+	if buf[0] != '\n' {
+		newLinesN++
+	}
+
+	lastPart := false
+
+Loop:
+	for {
+		if filePos < 0 {
+			filePos = 0
+			lastPart = true
+
+			buf = make([]byte, fileSize%bufSize)
+		}
+
+		if err := readFromPos(f, filePos, &buf); err != nil {
+			return 0, fmt.Errorf("%s", err)
+		}
+
+		for i := len(buf) - 1; i >= 0; i-- {
```

Comment:
Seems it could be moved into separate function

Reply:
This is the main part of this function, I'm not sure that is should be separated.